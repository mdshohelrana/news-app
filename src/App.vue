<template>
  <div id="app">
    <v-app id="news">
      <v-app-bar app color="indigo" flat>
        <v-container class="py-0 fill-height">
          <router-link to="/" class="router-link">
            <v-avatar color="white" size="45">N</v-avatar>
          </router-link>
          <v-btn
            class="text-capitalize headline white--text"
            to="/"
            v-for="link in links"
            :key="link"
            text
            plain
            style="text-decoration: none"
          >
            {{ link }}
          </v-btn>
          <v-spacer></v-spacer>
          <v-responsive max-width="260">
            <v-text-field
              v-model.trim="searchText"
              @keyup="search()"
              class="white"
              dense
              flat
              hide-details
              rounded
              solo-inverted
            ></v-text-field>
          </v-responsive>
        </v-container>
      </v-app-bar>
      <router-view />
    </v-app>
  </div>
</template>

<script>
import debounce from "lodash.debounce";

export default {
  data() {
    return {
      searchText: "",
      links: ["News"],
    };
  },

  created() {
    this.$store.dispatch("news/getTopHeadlines");
    this.$store.dispatch("news/getNewsSources");
  },

  methods: {
    search: debounce(function () {
      if (this.searchText.length > 0) {
        this.$store.dispatch("news/getSearchedTopHeadlines", this.searchText);
      } else {
        this.$store.dispatch("news/getTopHeadlines");
      }
    }, 500),
  },
};
</script>

<style>
#app {
  font-family: "Avenir", Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
#nav {
  padding: 30px;
}

#nav a {
  font-weight: bold;
  color: #2c3e50;
}

#nav a.router-link-exact-active {
  color: #42b983;
}

.router-link {
  text-decoration: none;
}

.v-application a {
  color: #000000 !important;
  text-decoration: none;
}

.v-application .read-more {
  color: #000000 !important;
  font-weight: bold;
}

.v-application a:hover {
  text-decoration: underline;
  cursor: pointer;
}

@media (min-width: 1904px) {
  .container {
    max-width: 1440px;
  }
}
</style>