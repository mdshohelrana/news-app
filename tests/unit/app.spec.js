// Libraries
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

// Components
import App from '@/App'

// Utilities
import {
  createLocalVue,
  shallowMount,
} from '@vue/test-utils'

Vue.use(Vuetify)
const localVue = createLocalVue()
localVue.use(Vuex)

describe('App.vue', () => {
  let vuetify
  let actions
  let store

  beforeEach(() => {
    actions = {
      "news/getTopHeadlines": jest.fn(),
      "news/getNewsSources": jest.fn(),
    }

    store = new Vuex.Store({
      state: {},
      actions
    })

    vuetify = new Vuetify()
  })

  it('should have a title and match snapshot', () => {
    const wrapper = shallowMount(App, {
      localVue,
      store,
      vuetify,
      stubs: ['router-link', 'router-view'],
    })

    expect(wrapper.html()).toMatchSnapshot()
    // expect(actions.SOME_ACTION).toHaveBeenCalled()

  })
})