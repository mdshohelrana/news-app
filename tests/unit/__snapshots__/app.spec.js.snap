// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should have a title and match snapshot 1`] = `
<div id="app">
  <v-app-stub id="news">
    <v-app-bar-stub color="indigo" tag="header" extensionheight="48" flat="true" src="" value="true" app="true">
      <v-container-stub tag="div" class="py-0 fill-height">
        <router-link-stub to="/" class="router-link">
          <v-avatar-stub color="white" size="45">N</v-avatar-stub>
        </router-link-stub>
        <v-btn-stub tag="button" activeclass="" to="/" plain="true" text="true" type="button" class="text-capitalize headline white--text" style="text-decoration: none;">
          News
        </v-btn-stub>
        <v-spacer-stub tag="div"></v-spacer-stub>
        <v-responsive-stub maxwidth="260">
          <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" dense="true" hidedetails="true" loaderheight="2" clearicon="$clear" flat="true" rounded="true" soloinverted="true" type="text" class="white"></v-text-field-stub>
        </v-responsive-stub>
      </v-container-stub>
    </v-app-bar-stub>
    <router-view-stub></router-view-stub>
  </v-app-stub>
</div>
`;
